### DISCUSSION TITLE
add Role to Member in TypeScript is not known

### DISCUSSION THREAD

**Discussion by Paradiesvogel7** [discordjs/discord.js#8413] [Q&A] [answered]
### Which package is this bug report for?

discord.js

### Issue description

Today I wanted to create a role menu. I wrote this in `TypeScirpt `and noticed that the method `roles.add()` is not recognised.

### Code sample

```typescript
if(interaction.isSelectMenu() && interaction.customId === "menu"){
interaction.member.roles.add(roleID);
}
```

### Package version

14.1.2

### Node.js version

18.6.0

### Operating system

Windows

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

GuildMember, Message

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildVoiceStates, GuildMessages

**Comment by almeidx** [comment, marked as answer]
Typeguard the interaction using `interaction.inCachedGuild()`, otherwise `interaction.member` could be an `APIGuildMember`, in which case `.roles` would be an array of snowflakes

**Reply by Paradiesvogel7** [reply]
So has someone an example?

**Reply by manuelvleeuwen** [reply]
As said, check if the guild in which the interaction occurs is cached, using the `inCachedGuild()` type guard on the interaction.

**Reply by Paradiesvogel7** [reply]
sorry i misunderstood you. now i have it, thanks
